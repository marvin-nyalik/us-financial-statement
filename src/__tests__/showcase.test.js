import { render } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import Showcase from '../components/showcase';

describe('Test the header component', () => {
  const mockStore = configureMockStore([thunk]);
  let store;
  beforeEach(() => {
    store = mockStore({
      company: {
        report: [
          {
            date: 12 / 2 / 1212,
            currency: 'USD',
            year: 2012,
            researchExpenses: 124934,
            income: 2394,
          },
          {
            date: 12 / 2 / 1213,
            currency: 'KES',
            year: 2013,
            researchExpenses: 1249334,
            income: 2394532,
          },
        ],
      },
      loading: false,
      companies: {
        companies: [],
      },
    });
  });

  it('Matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Showcase />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Renders the companies', () => {
    const store = mockStore({
      companies: {
        companies: ['AAPL', 'BACL'],
      },
    });

    const { getByText } = render(
      <Provider store={store}>
        <Showcase />
      </Provider>,
    );

    expect(getByText('AAPL')).toBeInTheDocument();
  });

  it('Displays all companies', () => {
    const store = mockStore({
      companies: {
        companies: ['AAPL', 'BACL'],
      },
    });

    const { getByText } = render(
      <Provider store={store}>
        <Showcase />
      </Provider>,
    );

    expect(getByText('AAPL')).toBeInTheDocument();
    expect(getByText('BACL')).toBeInTheDocument();
  });
});
