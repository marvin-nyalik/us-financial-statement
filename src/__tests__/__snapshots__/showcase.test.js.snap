// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the header component Matches the snapshot 1`] = `
<div>
  <div
    class="showcase"
  >
    <div
      class="showDiv"
    >
      <img
        alt="avatar"
        class="showImg"
      />
    </div>
    <div
      class="showDiv"
    >
      <h3
        class="showline"
      />
      <div
        class="companyList"
      />
    </div>
  </div>
</div>
`;
